### DISCUSSION TITLE
orienation in montage gets wrong

### DISCUSSION THREAD

**Discussion by estatistics** [ImageMagick/ImageMagick#6130] [Help] [answered]
### ImageMagick version

ImageMagick 6.9.11-60 Q16 x86_64 2021-01-25

### Operating system

Linux

### Operating system, version and so on

Debian GNU/Linux 11 (bullseye)

### Description

orienation in montage gets wrong

I remove almost all exif fields. Both photos, the vertical side is the biggest one eg. 4000x2000. However, montage gets 1st vertical and 2nd one horizontal. I cant explain that behavior. 

removing exif data
`find . -name '*.jpg'  -exec exiftool  -r -all=""  *.jpg *.jpeg *.JPG {} +`

montaging...
`montage aa_01.jpg aa_02.jpg -tile 2x1 -geometry +1+1  -quality 100 screenshot1.jpg`

Watch the result! 
![Επιλογή_120](https://user-images.githubusercontent.com/41580704/222780194-3216e1d8-8be6-48c0-ace2-0f85f24d5e93.jpg)

### Steps to Reproduce

removing exif data
`find . -name '*.jpg'  -exec exiftool  -r -all=""  *.jpg *.jpeg *.JPG {} +`

montaging...
`montage aa_01.jpg aa_02.jpg -tile 2x1 -geometry +1+1  -quality 100 screenshot1.jpg`

**Comment by estatistics** [comment]
note trying  `exiftool "-Y Cb Cr Sub Sampling=YCbCr4:2:2 (2 1)"  aa_02.jpg` results... 
Warning: Tag 'Y' is not defined

**Comment by fmw42** [comment]
Use -auto-orient on your photographs first.  You may have to use convert images -auto-orient miff:- | montage ...

That is pipe the result of convert ... -auto-orient to montage as I do not think montage knows about -auto-orient. But you can try using it directly in montage. If that fails then do the pipe.

Do not remove your exif info as the orientation flag is located there.

**Comment by estatistics** [comment]
Gimp information:  but displays one vertical and the other horizontal... is there any other hidden exif tag field?
![Επιλογή_122](https://user-images.githubusercontent.com/41580704/222784205-f797e339-056f-4a75-8a7d-4be13f4bd3d5.jpg)
![Επιλογή_121](https://user-images.githubusercontent.com/41580704/222784238-db7ffd7b-48e9-41fd-bbf7-fc9e3091eae5.jpg)

**Comment by estatistics** [comment]
> -auto-orient

i use it as:  `montage -auto-orient    -tile 2x1 -geometry +1+1    -quality 100   tiled.jpg`?

this did not work

**Comment by estatistics** [comment]
`convert *.jpg miff:-| montage -  -tile 2x1 -geometry +1+1  -quality 100 screen.jpg`

also, this one does not work

**Comment by fmw42** [comment]
Post your two input image.

**Comment by estatistics** [comment]
![a1](https://user-images.githubusercontent.com/41580704/222791919-8c10f40a-2612-4b51-a169-c8dcf83cc417.jpg)
![a2](https://user-images.githubusercontent.com/41580704/222791949-ff8c7bf7-0e1d-4b2e-b36a-93f746f360cb.jpg)

**Comment by estatistics** [comment]
photos with wetransfer: https://we.tl/t-Gd6F9WVMiG

**Comment by estatistics** [comment]
How i can tell imagemagick or motnage to rotate all pictures, with biggest side to be the vertical one?

**Comment by estatistics** [comment]
I wonder how imagemagick, and gimp and image viewer, display one image vertical and the other horizontal, when the width of both images is higher than the height of them, and in both images exif data cleared except a few?

**Comment by schumaml** [comment]
Either I am missing something obvious, or this is just like the images are. I mean, the width and height as displayed clearly indicate that one of them is about squarish, and the other much wider than high, and this is consistent with what we see.

What exactly makes you think that they should be displayed differently?

**Comment by estatistics** [comment]
a1: length= 2848, width=4288, a2: length = 4247 width = 4325 .. in both images width is higher than length, but all programs that i have used: imagemagick, gimp, image viewer, display one horizontal and the other vertical. on what condition? any hidden exif data ?

**Comment by fmw42** [comment]
Did you -strip these images?  The both have Orientation: Undefined.  So -auto-orient cannot process it.

You will have to write a script to find the largest dimension and rotate it. But you won't know whether to rotate +90 or -90. That is why the Orientation flag is important to keep when the camera takes the photo.

**Comment by estatistics** [comment]
ok. thanks . I think that imagemagick can include a parameter according to the largest side, to rotate images. This will be helpful for images with or without orientation tag (maybe), i think.

**Comment by fmw42** [comment]
You can do that easily in Imagemagick 7, but not Imagemagick 6. However even in Imagemagick 7, you won't know whether to rotate 90 or -90.

**Reply by estatistics** [reply]
How this can be done in imagemagick 7? Can you post an example?

**Comment by snibgo** [comment, marked as answer]
For example:
```
magick toes.png -rotate "%[fx:w>h?-90:0]" x.png
```
If the width is greater than the height, we rotate by -90 degrees (ie 90 degrees anti-clockwise). Otherwise we rotate by 0 degrees.
